components/datadog/agent/install.go

```diff
@@ -115,27 +116,36 @@ func updateAgentConfig(
 
 func installIntegrations(
 	fileManager *command.FileManager,
-	integrations map[string]string,
+	files map[string]string,
 	os os.OS,
 	lastCommand *remote.Command) (*remote.Command, string, error) {
 	configFolder := os.GetAgentConfigFolder()
 	var parts []string
-	for folderName, content := range integrations {
+	for filePath, content := range files {
 		var err error
-		folderPath := path.Join(configFolder, "conf.d", folderName)
-		confPath := path.Join(folderPath, "conf.yaml")
+		var fullPath string
+		useSudo := false
+		// filePath is absolute path from params.WithFile but relative from params.WithIntegration
+		if !strings.HasPrefix(filePath, "/") && !strings.HasPrefix(filePath, "C:\\") {
+			fullPath = path.Join(configFolder, filePath)
```

Comment:
I was hoping to move all this path building to `agentparams.WithIntegration(...)` but the param shouldn't have context around the os so os.GetAgentConfigFolder() isn't known at that time

Reply:
You can keep both `Integrations map[string]string` and `Files map[string]string`. Before calling this function you can fill `Files` with `Integrations` as you have access to `os.GetAgentConfigFolder()`. It should allow to remove this code. What do you think?

Reply:
Fixed in https://github.com/DataDog/test-infra-definitions/pull/289/commits/54784169e31df63e105a4f7862d913f738dc7b62